inc/class-sitemaps-categories.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * Class Core_Sitemaps_Categories.
+ * Builds the sitemap pages for Categories.
+ */
+class Core_Sitemaps_Categories extends Core_Sitemaps_Provider {
+	/**
+	 * Post type name.
+	 *
+	 * @var string
+	 */
+	protected $object_type = 'category';
+	/**
+	 * Sitemap name
+	 * Used for building sitemap URLs.
+	 *
+	 * @var string
+	 */
+	public $name = 'categories';
+
+	/**
+	 * Sitemap route.
+	 *
+	 * Regex pattern used when building the route for a sitemap.
+	 *
+	 * @var string
+	 */
+	public $route = '^sitemap-categories-?([0-9]+)?\.xml$';
+	/**
+	 * Sitemap slug.
+	 *
+	 * Used for building sitemap URLs.
+	 *
+	 * @var string
+	 */
+	public $slug = 'categories';
+
+	/**
+	 * Get a URL list for a user sitemap.
+	 *
+	 * @param string $object_type Name of the object_type.
+	 * @param int    $page_num Page of results.
+	 * @return array $url_list List of URLs for a sitemap.
+	 */
+	public function get_url_list( $object_type, $page_num = 1 ) {
```

Comment:
I've refactored mine to get rid of the object_type (which is $this->object-type, although not in this PR strangely enough), and made $page_num non-optional, as it's always checked before passing in.